### PR TITLE
Add React linting rules

### PR DESCRIPTION
**Pull request by pcardune**
Depending on how you have eslint configured with your editor, (if you use a globally installed eslint), you may need to run `npm install -g eslint-plugin-react`

### CODE REVIEW

File: apps/src/applab/designElements/ZOrderRow.jsx [hunk 1]
```diff
@@ -6,17 +6,7 @@ var FontAwesome = require('../../templates/FontAwesome');
...
-
-  getInitialState: function () {
-    return {
-      value: this.props.initialValue
-    };
-  },
-
-  componentWillReceiveProps: function (newProps) {
-    this.setState({value: newProps.initialValue});
+    onDepthChange: React.PropTypes.func.isRequired,
```
Comment (Bjvanminnen) [inline apps/src/applab/designElements/ZOrderRow.jsx:9]: Intentionally deleting these two methods?

File: apps/src/code-studio/components/share_dialog.jsx [hunk 2]
```diff
@@ -62,3 +62,20 @@ var ShareDialog = React.createClass({
...
+        {
+          name: 'basic example',
+          description: ``,
+          story: () => (
+            <ShareDialog />
+          )
+        },
+      ]);
+  };
+}
```
Comment (Bjvanminnen) [inline apps/src/code-studio/components/share_dialog.jsx]: Is this meant to be part of this PR? Seems like a separate thing

**Comment by Bjvanminnen** [comment]
Very cool. Thanks for finding all these propTypes :) Couple questions, otherwise lgtm

File: apps/src/applab/designElements/ZOrderRow.jsx (hunk 1 above)
Comment (pcardune) [inline apps/src/applab/designElements/ZOrderRow.jsx:9]: yeah, `this.state.value` never gets used. it's just copy-paste cruft.

**Pull request merged**